src/smif/cli/log.py

```diff
@@ -100,28 +69,50 @@ def summary(self, *args, **kws):
             self._log(logging.INFO, entry, args)
 
 
+def setup_logging(loglevel):
+    config = {
+        'version': 1,
+        'formatters': {
+            'default': {
+                'format': '%(asctime)s %(name)-12s: %(levelname)-8s %(message)s'
+            },
+            'message': {
+                'format': '\033[1;34m%(levelname)-8s\033[0m %(message)s'
+            }
+        },
+        'handlers': {
+            'file': {
+                'class': 'logging.FileHandler',
+                'level': 'DEBUG',
+                'formatter': 'default',
+                'filename': 'smif.log',
+                'mode': 'a',
+                'encoding': 'utf-8'
+            },
+            'stream': {
+                'class': 'logging.StreamHandler',
+                'formatter': 'message',
+                'level': 'DEBUG'
+            }
+        },
+        'root': {
+            'handlers': ['file', 'stream'],
+            'level': 'DEBUG'
+        }
+    }
+
+    if loglevel is None:
+        config['root']['level'] = logging.WARNING
+    elif loglevel == 1:
+        config['root']['level'] = logging.INFO
+    else:
+        config['root']['level'] = logging.DEBUG
+
+    logging.config.dictConfig(config)
+    logging.debug('Debug logging enabled.')
+
+
 logging.Logger.profiling_start = profiling_start
 logging.Logger.profiling_stop = profiling_stop
 logging.Logger.summary = summary
 logging.Logger._profile = OrderedDict()
-
-# Configure logging once, outside of any dependency on argparse
```

Comment:
Removing this block should be a quality+clarity improvement.

Reply:
Looks good @tomalrussell